.github/workflows/main.yml

```diff
@@ -907,10 +907,8 @@ jobs:
           java-version: 17
           distribution: ${{ env.default_java_distribution }}
 
-      # TODO fix JDK 21 incompatibilites
       - name: java/java.mx.project
-        continue-on-error: ${{ github.event_name != 'pull_request' }}
-        run: .github/retry.sh ant $OPTS -f java/java.mx.project test
+        run: ant $OPTS -f java/java.mx.project test
```

Comment:
@entlicher would you mind adding the retry script again? The mx project tests fail rarely, but they do fail sometimes as seen here:
https://github.com/apache/netbeans/actions/runs/7571474085
(scroll down and check annotations)

the build tools job has several unreliable steps which would have a negative effect on job stability.

Reply:
Sure, I'll add that.